src/test/c_api_tests.cpp

```diff
@@ -286,6 +288,151 @@ TEST_F(CAPIInference, TensorSetMovedBuffer) {
     ASSERT_EQ(tensor.setBuffer(std::move(buffer)), ovms::StatusCode::DOUBLE_BUFFER_SET);
 }
 
+TEST_F(CAPIInference, NoInputs) {
+    tensor_map_t m;
+    [[maybe_unused]] ovms::ServableMetadata sm("dummy", 1, m, m);
+    OVMS_ServableMetadata* osm = reinterpret_cast<OVMS_ServableMetadata*>(&sm);
+    uint32_t count;
+    ASSERT_EQ(sm.getVersion(), 1);
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServableMetadataGetInputCount(osm, &count));
+    ASSERT_EQ(count, 0);
+}
+
+TEST_F(CAPIInference, InferenceRequest) {
+    InferenceRequest* r = new InferenceRequest("dummy", 1);
+    size_t batchSize;
+    ASSERT_EQ(r->getBatchSize(batchSize, 1), StatusCode::INTERNAL_ERROR);
+    ASSERT_EQ(r->removeInputBuffer("dummy"), StatusCode::NONEXISTENT_TENSOR_FOR_REMOVE_BUFFER);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceRequestInputRemoveData(nullptr, "dummy"), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_InferenceRequestInputRemoveData(reinterpret_cast<OVMS_InferenceRequest*>(r), nullptr), StatusCode::NONEXISTENT_PTR);
+    delete r;
+}
+TEST_F(CAPIInference, InferenceResponse) {
+    InferenceResponse* r = new InferenceResponse("dummy", 1);
+    int64_t a = 1;
+    EXPECT_THROW(r->addOutput("name", OVMS_DataType::OVMS_DATATYPE_BIN, &a, -1), std::exception);
```

Comment:
Here we pass invalid pointer to addOutput and expecting exception throw. It will not always be the case since using invalid ptr ends in undefined behavior, including segfault/core dump.

You pass -1 to size_t dimCount which cannot be tested as well. This is overflow which we also are not able to validate in addOutput call.